public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/SimplifiedRuleEditor.test.tsx

```diff
@@ -0,0 +1,357 @@
+import { fetchRulerRulesIfNotFetchedYet } from 'app/features/alerting/unified/state/actions';
+import * as utils_config from 'app/features/alerting/unified/utils/config';
+import {
+  AlertManagerDataSource,
+  DataSourceType,
+  GRAFANA_DATASOURCE_NAME,
+  GRAFANA_RULES_SOURCE_NAME,
+  getAlertManagerDataSourcesByPermission,
+  useGetAlertManagerDataSourcesByPermissionAndConfig,
+} from 'app/features/alerting/unified/utils/datasource';
+import { getDefaultQueries } from 'app/features/alerting/unified/utils/rule-form';
+import { searchFolders } from 'app/features/manage-dashboards/state/actions';
+import { DashboardSearchHit } from 'app/features/search/types';
+import { AccessControlAction } from 'app/types';
+import { GrafanaAlertStateDecision, PromApplication } from 'app/types/unified-alerting-dto';
+
+import { RECEIVER_META_KEY } from '../../../contact-points/useContactPoints';
+import { ContactPointWithMetadata } from '../../../contact-points/utils';
+import { ExpressionEditorProps } from '../../ExpressionEditor';
+
+jest.mock('app/features/alerting/unified/components/rule-editor/ExpressionEditor', () => ({
+  // eslint-disable-next-line react/display-name
+  ExpressionEditor: ({ value, onChange }: ExpressionEditorProps) => (
+    <input value={value} data-testid="expr" onChange={(e) => onChange(e.target.value)} />
+  ),
+}));
+
+jest.mock('app/features/alerting/unified/api/buildInfo');
+jest.mock('app/features/alerting/unified/api/ruler');
+jest.mock('app/features/manage-dashboards/state/actions');
+
+jest.mock('app/core/components/AppChrome/AppChromeUpdate', () => ({
+  AppChromeUpdate: ({ actions }: { actions: React.ReactNode }) => <div>{actions}</div>,
+}));
+
+// there's no angular scope in test and things go terribly wrong when trying to render the query editor row.
+// lets just skip it
+jest.mock('app/features/query/components/QueryEditorRow', () => ({
+  // eslint-disable-next-line react/display-name
+  QueryEditorRow: () => <p>hi</p>,
+}));
+
+// simplified routing mocks
+const grafanaAlertManagerDataSource: AlertManagerDataSource = {
+  name: GRAFANA_RULES_SOURCE_NAME,
+  imgUrl: 'public/img/grafana_icon.svg',
+  hasConfigurationAPI: true,
+};
+jest.mock('app/features/alerting/unified/utils/datasource', () => {
+  return {
+    ...jest.requireActual('app/features/alerting/unified/utils/datasource'),
+    getAlertManagerDataSourcesByPermission: jest.fn(),
+    useGetAlertManagerDataSourcesByPermissionAndConfig: jest.fn(),
+    getAlertmanagerDataSourceByName: jest.fn(),
+  };
+});
+
+const user = userEvent.setup();
+
+jest.spyOn(utils_config, 'getAllDataSources');
+jest.spyOn(dsByPermission, 'useAlertManagersByPermission');
+jest.spyOn(useContactPoints, 'useContactPointsWithStatus');
+
+jest.setTimeout(60 * 1000);
+
+const mocks = {
+  getAllDataSources: jest.mocked(utils_config.getAllDataSources),
+  searchFolders: jest.mocked(searchFolders),
+  useContactPointsWithStatus: jest.mocked(useContactPoints.useContactPointsWithStatus),
```

Comment:
Instead of mocking `useContactPoints` can we use the `setupGrafanaManagedServer` mock which uses MSW instead of mocking the hooks?